### PR TITLE
Adds Craft Sub-Generator

### LINKED ISSUE #32: Create a "craft" sub-generator that will install and configure craft.
I was originally thinking we should use https://github.com/nystudio107/generator-craftinstall for this, but after taking another look at the project it does a lot of things we wouldn't have a need for, and would be pretty difficult to compose with.

### PR DESCRIPTION
**Pull request by cmalven**
Closes #32 

Adds a new sub-generator responsible for everything related to setting up Craft for a new project:

- Downloads Craft
- Installs common Craft plugins (giving you the option to exclude any you don't want) via Composer when possible, or by downloading the plugin from Github otherwise.
- Sets up `.env`-based environment config setup and configures everything necessary on the Craft side.
- Cleans out default template files and adds some nice, simple default templates.
- Sets appropriate default permissions on all Craft directories.

There is a lot going on here, so I'd appreciate anyone who can running this and seeing if there is anything missing.

### CODE REVIEW

File: generators/craft/index.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        fs.renameSync(pluginDirPath, self.destinationPath('craft/plugins/' + option));
+      });
+      del.sync([self.destinationPath('craft/plugins/downloads/')]);
+    },
+
+    permissions: function() {
+      console.log(chalk.green('> Setting global permissions to 755 / 644'));
+      child_process.execSync('chmod -R 755 *');
+      child_process.execSync('find . -type f -exec chmod 644 {} \\;');
+      console.log(chalk.green('> Setting permissions on craft/app to 775 / 644'));
```
Comment (mkornatz) [inline generators/craft/index.js:172]: Is there a reason you're changing perms on app, config, and storage separately? I imagine the first two lines should take care of all sub directories of the project.

Comment (cmalven) [reply generators/craft/index.js:172]: I copied the setting of these permissions directly from https://github.com/nystudio107/generator-craftinstall/blob/master/app/index.js#L299 and assumed he had a good reason for doing it the way he is.

Comment (mkornatz) [reply generators/craft/index.js:172]: Hmm, I'm not sure why he has these there. Lines 168 through 176 really have no effect after 166 and 167 run (which are already recursive). You should be able to safely remove the lines after 167.

**Comment by mkornatz** [comment]
This looks really good. I can't think of anything else to add at the moment. Great work!

File: generators/craft/index.js (hunk 1 above)
Comment (cmalven) [reply generators/craft/index.js:172]: Updated

Comment (mkornatz) [reply generators/craft/index.js:172]: Oh, I missed something critical! He's using 644 (globally) and 664 for the individual dirs. That gives group write permissions, which is a good idea. Disregard my previous comment.

**Comment by brianjhanson** [comment]
Just tried to run this and got:
```
Error: Cannot find module 'download'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/brianhanson/Development/generator-one-base/generators/craft/index.js:5:16)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
```

I think this is happening because it's trying to run the Craft generator before actually installing the modules.


EDIT: This might be my fault. Please hold.

**Comment by cmalven** [comment]
@brianjhanson make sure you run `npm install && npm link` in the `generator-one-base` directory before trying to run the generator.

**Comment by brianjhanson** [comment]
Yep, realized that shortly after posting.

**Comment by brianjhanson** [comment]
Seemed to work well.

**Pull request merged**